evap/staff/tests/test_views.py

```diff
@@ -2349,6 +2381,19 @@ def test_state_change_log_translated(self, trans):
         response = self.app.get(self.url, user=self.manager)
         self.assertContains(response, "TRANSLATED-state: TRANSLATED-new &#8594; TRANSLATED-prepared")
 
+    def test_evaluation_confirm_without_main_language(self):
+        response = self.app.get(self.url, user=self.manager, status=200)
+        form = response.forms["evaluation-form"]
+
+        form["main_language"] = "x"
+        response = form.submit("operation", value="approve")
+        self.assertContains(response, "You have to set a main language to approve this evaluation.")
+
+        self.assertNotEqual(Evaluation.objects.first().state, self.evaluation.State.APPROVED)
+
+        form["main_language"] = "en"
+        form.submit("operation", value="approve")
```

Comment:
We probably have to add an assertion that the response now doesn't contain the above error anymore (our views return 200 even if there is a business logic error / form validation failed)

Reply:
Which error do you mean?

Reply:
> You have to set a main language to approve this evaluation.")

from line 2437